NitroxPatcher/Patches/Dynamic/IngameMenu_OnDeselect_Patch.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Reflection;
+using HarmonyLib;
+using NitroxClient.MonoBehaviours.Gui.InGame;
+using NitroxModel.Helper;
+
+namespace NitroxPatcher.Patches.Dynamic;
+
+/// <summary>
+/// Disable the possibility of escaping a modal by clicking outside of its box or pressing the escape for the modals that want it
+/// </summary>
+public class IngameMenu_OnDeselect_Patch : NitroxPatch, IDynamicPatch
+{
+    private static MethodInfo TARGET_METHOD = Reflect.Method((IngameMenu t) => t.OnDeselect());
+
+    // OnDeselect happens when you deselect the modal (by clicking outside or pressing escape)
+    // Therefore, if we prevent it from happening it certain cases, it will force the user to click one of the buttons of the modal
+    public static bool Prefix(IngameMenu __instance)
+    {
+        // Cancel if the modal is set to be non-avoidable (IsAvoidable) and if it's not the modal that wants to close itself (IsAvoidableBypass)
+        if (Modal.TryGetByName(__instance.currentScreen.name, out Modal modal) && !modal.IsAvoidable && !modal.IsAvoidableBypass)
```

Comment:
What gamobject is `__instance.currentScreen` exactly? Could we use `GetComponent<Modal>` to get the current Modal?

Reply:
they're just the visual modal, the component is located in the Multiplayer object

Reply:
Mhh can we have a static "currentModal" field in `Modal.cs`? I would only use the string to type conversation if really necessary.

Reply:
ok I'll go for it